### ISSUE TITLE
App crashed when trying to change save location

### ISSUE THREAD

**Issue by tim77** [SeaDve/Kooha#122]
**System Info**
- Kooha version 2.0.0
- Distro information and version Fedora 34 Workstation
- Desktop Environment GNOME 40
- Display Server Xorg
- Flatpak? Yes.

**Describe the bug**
App crashed when trying to change save location:

```rs
thread 'main' panicked at 'assertion failed: !ptr.is_null()', /run/build/kooha/vendor/glib/src/translate.rs:1508:9
```

`xdg-desktop-portal-wlr` installed.

**Comment by SeaDve** [comment]
Not sure where that's coming from, but you can `RUST_BACKTRACE=1` to show the full backtrace

**Comment by tim77** [comment]
This with `RUST_BACKTRACE=1` and unfortunately i don't see any additional info there. Maybe this happened only on localised version? Didn't tested EN version yet.

Also my location for Videos changed to external hard drive via `xdg-user-dirs-update`. Maybe this relates somehow.

**Comment by SeaDve** [comment]
`RUST_BACKTRACE=1` would be helpful for me where exactly the program crashed

**Comment by tim77** [comment]
I've got one with `RUST_BACKTRACE=full`:

<details>
  <summary>Backtrace</summary>
    
```rs
stack backtrace:
   0:     0x5618731468b0 - <unknown>
   1:     0x56187316668c - <unknown>
   2:     0x561873143725 - <unknown>
   3:     0x5618731489bb - <unknown>
   4:     0x561873148491 - <unknown>
   5:     0x561873149084 - <unknown>
   6:     0x561873148b37 - <unknown>
   7:     0x561873146dac - <unknown>
   8:     0x561873148ac9 - <unknown>
   9:     0x561872ef41c1 - <unknown>
  10:     0x561872ef410d - <unknown>
  11:     0x56187312e95f - <unknown>
  12:     0x561872f93e6d - <unknown>
  13:     0x561872f40f3e - <unknown>
  14:     0x7f9327162fef - g_closure_invoke
  15:     0x7f9327175e66 - <unknown>
  16:     0x7f932717cb5c - g_signal_emit_valist
  17:     0x7f932717ccb3 - g_signal_emit
  18:     0x7f9326466ed5 - <unknown>
  19:     0x7f9326adbae7 - <unknown>
  20:     0x7f9326ad7a18 - <unknown>
  21:     0x7f9327162fef - g_closure_invoke
  22:     0x7f9327175e66 - <unknown>
  23:     0x7f932717cb5c - g_signal_emit_valist
  24:     0x7f932717ccb3 - g_signal_emit
  25:     0x7f93268068ef - <unknown>
  26:     0x7f9327163229 - <unknown>
  27:     0x7f932717ca98 - g_signal_emit_valist
  28:     0x7f932717ccb3 - g_signal_emit
  29:     0x7f93268e59c6 - <unknown>
  30:     0x7f932716648f - g_cclosure_marshal_VOID__BOXEDv
  31:     0x7f9327163229 - <unknown>
  32:     0x7f932717ca98 - g_signal_emit_valist
  33:     0x7f932717ccb3 - g_signal_emit
  34:     0x7f93268e2277 - <unknown>
  35:     0x7f93268e3b03 - <unknown>
  36:     0x7f93268e7020 - <unknown>
  37:     0x7f93268b16da - <unknown>
  38:     0x7f9326a5b1ca - <unknown>
  39:     0x7f9326a62e8c - <unknown>
  40:     0x7f9326933686 - <unknown>
  41:     0x7f9326933d73 - <unknown>
  42:     0x7f9326963ce0 - <unknown>
  43:     0x7f9326b64487 - <unknown>
  44:     0x7f9326b8bee3 - <unknown>
  45:     0x7f9327162fef - g_closure_invoke
  46:     0x7f9327175e66 - <unknown>
  47:     0x7f932717c43b - g_signal_emit_valist
  48:     0x7f932717ccb3 - g_signal_emit
  49:     0x7f9326b8f554 - <unknown>
  50:     0x7f9326bdd6ce - <unknown>
  51:     0x7f9327066681 - g_main_context_dispatch
  52:     0x7f9327066b68 - <unknown>
  53:     0x7f9327066c33 - g_main_context_iteration
  54:     0x7f932645dc65 - g_application_run
  55:     0x561872f49ed1 - <unknown>
  56:     0x561872f4043e - <unknown>
  57:     0x561872fbb333 - <unknown>
  58:     0x561872fb0609 - <unknown>
  59:     0x56187314958a - <unknown>
  60:     0x561872f40722 - main
  61:     0x7f9325ec0b90 - __libc_start_main
  62:     0x561872ef499e - <unknown>
  63:                0x0 - <unknown>
```  

</details>

**Comment by SeaDve** [comment]
thanks! seems like debugging info is missing on release builds

**Comment by SeaDve** [comment]
https://github.com/SeaDve/Kooha/suites/3868846620/artifacts/96234104

Here is the debug build that should contain the additional info

**Comment by tim77** [comment]
Tried with `kooha-devel.flatpak` but no additional info here. I guess we need to try with full `<runtime-id>.Debug`. I'll try ASAP, this is huge piece of software which need to download.

**Comment by SeaDve** [comment]
I added debug info there about available cursor modes in the logs

**Comment by tim77** [comment]
With `RUST_BACKTRACE=full`:

<details>
  <summary>Backtrace io.github.seadve.Kooha.Devel</summary>

```rs
0:00:00.107731406     4 0x561cd5139290 WARN              vaapiblend gstvaapiblend.c:184:gst_vaapi_blend_initialize:<vaapiblend0> VPP does not support global alpha blending
0:00:00.250649112     4 0x561cd5139290 WARN                 nvcodec gstcudaloader.c:114:gst_cuda_load_library: Could not open library libcuda.so.1, libcuda.so.1: cannot open shared object file: No such file or directory
0:00:00.250661003     4 0x561cd5139290 WARN                 nvcodec plugin.c:66:plugin_init: Failed to load cuda library
[I][70151.882414][      pipewire.c:  562 pw_init()] version 0.3.34
0:00:00.818666122     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x120000: 'AVR (Audio Visual Research)' is not mapped
0:00:00.818697112     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x180000: 'CAF (Apple Core Audio File)' is not mapped
0:00:00.818718672     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x100000: 'HTK (HMM Tool Kit)' is not mapped
0:00:00.818745022     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xc0000: 'MAT4 (GNU Octave 2.0 / Matlab 4.2)' is not mapped
0:00:00.818771213     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xd0000: 'MAT5 (GNU Octave 2.1 / Matlab 5.0)' is not mapped
0:00:00.818791093     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x210000: 'MPC (Akai MPC 2k)' is not mapped
0:00:00.818817103     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0xe0000: 'PVF (Portable Voice Format)' is not mapped
0:00:00.818845123     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x160000: 'SD2 (Sound Designer II)' is not mapped
0:00:00.818872924     4 0x561cd5139290 WARN                 default gstsf.c:97:gst_sf_create_audio_template_caps: format 0x190000: 'WVE (Psion Series 3)' is not mapped
 INFO  kooha::application > Kooha (io.github.seadve.Kooha.Devel)
 INFO  kooha::application > Version: 2.0.0-8225d4f (Devel)
 INFO  kooha::application > Datadir: /app/share/kooha
thread 'main' panicked at 'assertion failed: !ptr.is_null()', /run/build/kooha/_flatpak_build/cargo-home/registry/src/github.com-1ecc6299db9ec823/glib-0.14.5/src/translate.rs:1508:9
stack backtrace:
   0:     0x56057c60f290 - <unknown>
   1:     0x56057c62f06c - <unknown>
   2:     0x56057c60c105 - <unknown>
   3:     0x56057c61139b - <unknown>
   4:     0x56057c610e71 - <unknown>
   5:     0x56057c611a64 - <unknown>
   6:     0x56057c611517 - <unknown>
   7:     0x56057c60f78c - <unknown>
   8:     0x56057c6114a9 - <unknown>
   9:     0x56057bfa66a1 - <unknown>
  10:     0x56057bfa65ed - <unknown>
  11:     0x56057c5d32c1 - <unknown>
  12:     0x56057c5a9146 - <unknown>
  13:     0x56057c5afaa1 - <unknown>
  14:     0x56057c06e29c - <unknown>
  15:     0x56057bfa87bf - <unknown>
  16:     0x56057bfce2d8 - <unknown>
  17:     0x56057c09a2ef - <unknown>
  18:     0x7f643c891fef - g_closure_invoke
  19:     0x7f643c8a4e66 - <unknown>
  20:     0x7f643c8abb5c - g_signal_emit_valist
  21:     0x7f643c8abcb3 - g_signal_emit
  22:     0x7f643bb93ed5 - <unknown>
  23:     0x7f643c208ae7 - <unknown>
  24:     0x7f643c204a18 - <unknown>
  25:     0x7f643c891fef - g_closure_invoke
  26:     0x7f643c8a4e66 - <unknown>
  27:     0x7f643c8abb5c - g_signal_emit_valist
  28:     0x7f643c8abcb3 - g_signal_emit
  29:     0x7f643bf338ef - <unknown>
  30:     0x7f643c892229 - <unknown>
  31:     0x7f643c8aba98 - g_signal_emit_valist
  32:     0x7f643c8abcb3 - g_signal_emit
  33:     0x7f643c0129c6 - <unknown>
  34:     0x7f643c89548f - g_cclosure_marshal_VOID__BOXEDv
  35:     0x7f643c892229 - <unknown>
  36:     0x7f643c8aba98 - g_signal_emit_valist
  37:     0x7f643c8abcb3 - g_signal_emit
  38:     0x7f643c00f277 - <unknown>
  39:     0x7f643c010b03 - <unknown>
  40:     0x7f643c014020 - <unknown>
  41:     0x7f643bfde6da - <unknown>
  42:     0x7f643c1881ca - <unknown>
  43:     0x7f643c18fe8c - <unknown>
  44:     0x7f643c060686 - <unknown>
  45:     0x7f643c060d73 - <unknown>
  46:     0x7f643c090ce0 - <unknown>
  47:     0x7f643c291487 - <unknown>
  48:     0x7f643c2b8ee3 - <unknown>
  49:     0x7f643c891fef - g_closure_invoke
  50:     0x7f643c8a4e66 - <unknown>
  51:     0x7f643c8ab43b - g_signal_emit_valist
  52:     0x7f643c8abcb3 - g_signal_emit
  53:     0x7f643c2bc554 - <unknown>
  54:     0x7f643c30a6ce - <unknown>
  55:     0x7f643c795681 - g_main_context_dispatch
  56:     0x7f643c795b68 - <unknown>
  57:     0x7f643c795c33 - g_main_context_iteration
  58:     0x7f643bb8ac65 - g_application_run
  59:     0x56057bfcdd22 - <unknown>
  60:     0x56057bfcddec - <unknown>
  61:     0x56057bfaa2af - <unknown>
  62:     0x56057c0867cc - <unknown>
  63:     0x56057bffa27b - <unknown>
  64:     0x56057c0a343e - <unknown>
  65:     0x56057c011401 - <unknown>
  66:     0x56057c611f6a - <unknown>
  67:     0x56057c0113d0 - <unknown>
  68:     0x56057c08683c - main
  69:     0x7f643b5ebb90 - __libc_start_main
  70:     0x56057bfa6e9e - <unknown>
  71:                0x0 - <unknown>
</details>

**Comment by Tuxino88** [comment]
Same problem by compiling from source on Fedora 34 WS

<details>
  <summary>crash on startup</summary>

(kooha:5069): Gtk-CRITICAL **: 14:39:24.792: Error building template class 'KoohaMainWindow' for an instance of type 'KoohaMainWindow': .:0:0 Invalid property: GtkMenuButton.primary
thread 'main' panicked at 'assertion failed: !self.ptr.is_null()', /builddir/build/BUILD/Kooha/x86_64-redhat-linux-gnu/cargo-home/registry/src/github.com-1ecc6299db9ec823/gtk4-0.3.0/src/subclass/widget.rs:1284:13
stack backtrace:
   0:     0x556b3820c3e0 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h77bd492c12b998b9
   1:     0x556b3822c4ec - core::fmt::write::hb6b225bcb934036e
   2:     0x556b38208f75 - std::io::Write::write_fmt::h418376ca7e3b9754
   3:     0x556b3820e53b - std::panicking::default_hook::{{closure}}::h35dd68e72cc15a04
   4:     0x556b3820e011 - std::panicking::default_hook::hc58f730297711b6a
   5:     0x556b3820ec14 - std::panicking::rust_panic_with_hook::hee0d54709bd44f21
   6:     0x556b3820e6b7 - std::panicking::begin_panic_handler::{{closure}}::h272a6437e18c53be
   7:     0x556b3820c8dc - std::sys_common::backtrace::__rust_end_short_backtrace::he9d3afb9a0922f4f
   8:     0x556b3820e649 - rust_begin_unwind
   9:     0x556b37fa2281 - core::panicking::panic_fmt::h05331ed052e7a23a
  10:     0x556b37fa21cd - core::panicking::panic::h0e8cddb1c5171c02
  11:     0x556b38043878 - kooha::widgets::main_window::MainWindow::setup_signals::hb7ceea2b0cbdd010
  12:     0x556b38078922 - glib::subclass::object::constructed::hb7ee9c5705e2ac99
  13:     0x7f54d2d68c77 - <unknown>
  14:     0x7f54d2d69f3a - g_object_newv
  15:     0x556b381f23ec - glib::object::Object::new_internal::hcd6ce196f919ec53
  16:     0x556b381f1ff5 - glib::object::Object::with_type::ha619bc3f131ef26c
  17:     0x556b3804345c - kooha::widgets::main_window::MainWindow::new::h4327db508cbcf2f3
  18:     0x556b37fe018d - <kooha::application::imp::Application as gio::subclass::application::ApplicationImpl>::activate::hfe3f1c97160e6362
  19:     0x556b3807bf42 - gio::subclass::application::application_activate::h1e1edb36604bc36e
  20:     0x7f54d2d7783a - g_signal_emit_valist
  21:     0x7f54d2d77983 - g_signal_emit
  22:     0x7f54d217d758 - <unknown>
  23:     0x556b3807c4b7 - gio::subclass::application::application_local_command_line::hed176faeb8ccac36
  24:     0x7f54d217d936 - g_application_run
  25:     0x556b380277a1 - kooha::application::Application::run::hcfcafded9ba16fdd
  26:     0x556b3801da9e - kooha::main::hd188d23d1855e989
  27:     0x556b38076b03 - std::sys_common::backtrace::__rust_begin_short_backtrace::h49d2ddc56da07330
  28:     0x556b380596e9 - std::rt::lang_start::{{closure}}::h81703bbf54534bf3
  29:     0x556b3820f11a - std::rt::lang_start_internal::hce070b7cec700fe0
  30:     0x556b3801dd82 - main
  31:     0x7f54d1bfdb75 - __libc_start_main
  32:     0x556b37fa2a5e - _start
  33:                0x0 - <unknown>

</details>

**Comment by SeaDve** [comment]
> Same problem by compiling from source on Fedora 34 WS

This seems not related with this issue. I think you need a newer version of gtk, specifically gtk 4.4 and above

**Comment by tim77** [comment]
> newer version of gtk, specifically gtk 4.4 and above

Host version should been irrelevant since we using flatpak build.

**Comment by Tuxino88** [comment]
> > Same problem by compiling from source on Fedora 34 WS
> 
> This seems not related with this issue. I think you need a newer version of gtk, specifically gtk 4.4 and above

ah OK thank you very much :)

**Comment by Tuxino88** [comment]
> It's seem like i found from where this issue comes. @Tuxino88 can you show your [1] please?
> 
> [1]:
> 
> ```shell
> $ cat $HOME/.config/user-dirs.dirs | fpaste
> ```

My problem was the version of gtk4, I have no problems now

**Comment by jahed** [comment]
@tim77 Just to confirm, you're right about user-dirs. Mine was set to:

```
XDG_VIDEOS_DIR="$HOME/"
```

I changed it to

```
XDG_VIDEOS_DIR="$HOME/Videos"
```

and it works now. Guessing because the flatpak environment only allows access to Videos.

**Comment by SeaDve** [comment]
likely a glib bug https://gitlab.gnome.org/GNOME/glib/-/issues/2529

**Comment by SeaDve** [comment]
I think this should be fixed in gtk4-rs 0.4.0 release

**Comment by SeaDve** [comment]
Closing this as fixed. However, feel free to reopen it if it still exists on main branch. Thanks!
